.circleci/config.yml

```diff
@@ -54,18 +55,98 @@ jobs:
           echo 'export SKIP_METRICS=y' >> $BASH_ENV
           echo 'export STAGE_TIMEOUT_MIN=30' >> $BASH_ENV
           echo 'export NUM_OF_RETRIES=2' >> $BASH_ENV
-          echo 'export TEST_CONFIG=test/acse-conf/acse-feature-validation.json' >> $BASH_ENV
-          echo 'export SUBSCRIPTION_ID=${SUBSCRIPTION_ID_FEATURE_VALIDATION}' >> $BASH_ENV
-          echo 'export SERVICE_PRINCIPAL_CLIENT_ID=${SERVICE_PRINCIPAL_CLIENT_ID_FEATURE_VALIDATION}' >> $BASH_ENV
-          echo 'export SERVICE_PRINCIPAL_CLIENT_SECRET=${SERVICE_PRINCIPAL_CLIENT_SECRET_FEATURE_VALIDATION}' >> $BASH_ENV
-          echo 'export CLUSTER_SERVICE_PRINCIPAL_CLIENT_ID=${CLUSTER_SERVICE_PRINCIPAL_CLIENT_ID_FEATURE_VALIDATION}' >> $BASH_ENV
-          echo 'export CLUSTER_SERVICE_PRINCIPAL_CLIENT_SECRET=${CLUSTER_SERVICE_PRINCIPAL_CLIENT_SECRET_FEATURE_VALIDATION}' >> $BASH_ENV
+          echo 'export SUBSCRIPTION_ID=${SUBSCRIPTION_ID_E2E_KUBERNETES}' >> $BASH_ENV
+          echo 'export SERVICE_PRINCIPAL_CLIENT_ID=${SERVICE_PRINCIPAL_CLIENT_ID_E2E_KUBERNETES}' >> $BASH_ENV
+          echo 'export SERVICE_PRINCIPAL_CLIENT_SECRET=${SERVICE_PRINCIPAL_CLIENT_SECRET_E2E_KUBERNETES}' >> $BASH_ENV
+          echo 'export CLUSTER_SERVICE_PRINCIPAL_CLIENT_ID=${SERVICE_PRINCIPAL_CLIENT_ID_E2E_KUBERNETES}' >> $BASH_ENV
+          echo 'export CLUSTER_SERVICE_PRINCIPAL_CLIENT_SECRET=${SERVICE_PRINCIPAL_CLIENT_SECRET_E2E_KUBERNETES}' >> $BASH_ENV
+          echo 'export TEST_CONFIG=test/e2e/kubernetes-deployments.json' >> $BASH_ENV
       - run:
-          name: Feature validation tests
+          name: Kubernetes feature validation tests
+          command: make bootstrap build test-e2e
+      - store_artifacts:
+          path: /go/src/github.com/Azure/acs-engine/_logs
+  master-swarm-e2e:
+    working_directory: /go/src/github.com/Azure/acs-engine
+    no_output_timeout: "30m"
+    docker:
+      - image: microsoft/acs-engine:latest
```

Comment:
> Does it hold the binary we are testing?

The image at the moment provides 1) a go dev environment and 2) the `az`CLI tool installed. Circle launches the build into a container with this image, then mounts the current source code of the branch into it, runs bootstrap, builds binaries, etc. 

I want to build immutable Docker artifacts for each build and run tests in that, but I couldn't figure out how to cache and reuse the built image between CI stages. I think it's possible, but takes some reading. Without caching a built image we'd add 3-4 minutes on every stage, so opted for the current approach in the interim.